src/Microsoft.Extensions.Hosting/Internal/Host.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.Hosting.Internal
+{
+    internal class Host : IHost
+    {
+        private ILogger<Host> _logger;
+        private IHostLifetime _hostLifetime;
+        private ApplicationLifetime _applicationLifetime;
+        private IEnumerable<IHostedService> _hostedServices;
+
+        internal Host(IServiceProvider services)
+        {
+            Services = services ?? throw new ArgumentNullException(nameof(services));
+            _applicationLifetime = Services.GetRequiredService<IApplicationLifetime>() as ApplicationLifetime;
+            _logger = Services.GetRequiredService<ILogger<Host>>();
+            _hostLifetime = Services.GetService<IHostLifetime>() ?? new NoOpHostLifetime();
```

Comment:
Why is this optional?

Reply:
IHost.Start/StopAsync can function fine without it.

Reply:
Or should I pre-register NoOpHostLifetime?

Reply:
I think the default would be the AppDomain.ProcessExit one.